lib/iris/coords.py

```diff
@@ -1580,29 +1652,28 @@ def _new_points_requirements(self, points):
         if len(points) > 1 and not iris.util.monotonic(points, strict=True):
             raise ValueError('The points array must be strictly monotonic.')
 
-    @property
-    def points(self):
-        """The local points values as a read-only NumPy array."""
-        return self._points_dm.data
-
-    @points.setter
-    def points(self, points):
-        # We must realise points to check monotonicity.
+    def _points_setter(self, points):
+        # DimCoord always realises the points, to allow monotonicity checks.
         copy = is_lazy_data(points)
         points = as_concrete_data(points)
-        points = np.array(points, copy=copy, ndmin=1)
+        # Ensure it is an array..
+        # .. and always a distinct view, so that we can make it read-only.
+        points = np.array(points, copy=copy)
 
-        # Set or update DataManager.
-        if self._points_dm is None:
-            self._points_dm = DataManager(points)
-        else:
-            self._points_dm.data = points
+        # Invoke the generic points setter.
+        super(DimCoord, self)._points_setter(points)
+
+        if self._points_dm is not None:
+            # Fetch the new core array, as the parent can change it.
+            points = self._points_dm._real_array
```

Comment:
I'm not at all a fan of accessing this private `DataManager` method. Surely there is a better way of doing this?